internal/scheduler/legacyscheduler.go

```diff
@@ -474,205 +585,127 @@ func (sched *LegacyScheduler) String() string {
 }
 
 func NewLegacyScheduler(
-	schedulingConfig configuration.SchedulingConfig,
-	executorId string,
-	totalResources schedulerobjects.ResourceList,
-	nodes []*schedulerobjects.Node,
+	ctx context.Context,
+	constraints SchedulingConstraints,
+	config configuration.SchedulingConfig,
+	nodeDb *NodeDb,
 	jobRepository SchedulerJobRepository,
 	priorityFactorByQueue map[string]float64,
+	initialResourcesByQueueAndPriority map[string]schedulerobjects.QuantityByPriorityAndResourceType,
 ) (*LegacyScheduler, error) {
-	priorities := make([]int32, 0)
-	cumulativePriorityLimits := make(map[int32]map[string]float64, 0)
-	for _, priority := range schedulingConfig.Preemption.PriorityClasses {
-		priorities = append(priorities, priority.Priority)
-		cumulativePriorityLimits[priority.Priority] = priority.MaximalResourceFractionPerQueue
+	if ResourceListAsWeightedApproximateFloat64(constraints.ResourceScarcity, constraints.TotalResources) == 0 {
```

Comment:
I'm not in an ide here so it's a bit difficuly to check, but comparing the rsults of a function  called `ResourceListAsWeightedApproximateFloat64` with an integer (in this case 0) seems wrong.

Reply:
The compiler should understand that the `0` is of type `float64`. I don't think there are any floating point comparison issues re comparing with 0.